### ISSUE TITLE
When is PPI width << classical width?

### ISSUE THREAD

**Issue by sms1097** [aangelopoulos/prediction-powered-inference#1]
I made an adaption of the ballots notebook for a similar binary problem (figuring out what the true percentage of happy customers are for a business.) In my case I find that the difference in width of the confidence intervals is often negligible. Is there a particular ratio of labeled to unlabeled data that improves these results? Can accuracy of the predictions make a difference?

**Comment by aangelopoulos** [comment]
Yeah, the accuracy of the predictions can have a big effect. Another aspect is the variance of $Y$ -- for example, if 99% of your customers are satisfied, it will be harder to get an interval that's better than the standard/classical interval that only uses the labeled data.

If you send along your model accuracy and also the fraction of $Y$s that are 0 and 1, respectively, then maybe I can help you understand whether this is expected behavior?

**Comment by aangelopoulos** [comment]
Oh also --- have you seen the new repo? https://github.com/aangelopoulos/ppi_py

**Comment by sms1097** [comment]
Did not see the new repo, this one actually shows up first in search so I'll check that out! 

Here are the counts I have for each variable. These are customer reviews for an airline, which means customers do tend to only leave negative reviews which shows here: 
y_labeled (array([0, 1]), array([172,  78]))
yhat_labeled (array([0., 1.]), array([176,  74]))
yhat_unlabeled (array([0., 1.]), array([3106, 1045]))

In my tests I've seen ~90% accuracy with the model I'm using. 

My goal was to see if I could use a small LLM with zero-shot classification to construct better confidence intervals than if I had only used human data or a more expensive LLM. The closest research I've found doing something like this is ARES, so if you have another resource you could point me in the direction of or any anecdotal thoughts those would be appreciated.

Here's the figure I created for my test:

![image](https://github.com/aangelopoulos/prediction-powered-inference/assets/54160079/3b109eb3-9ad4-4ddd-9e44-04531233537e)

**Comment by aangelopoulos** [comment]
Oh, cool!
Looks like it's working pretty well, except maybe the first point with width 0.  
Is this using the new library? U should use the new one, it should have better performance! :)

**Comment by aangelopoulos** [comment]
Regarding inference with LLMs, I really haven't seen anything like that, but it seems like a great application. I'll be interested in seeing your paper :)

**Comment by aangelopoulos** [comment]
(At Tijana's suggestion, here's a reference! https://arxiv.org/abs/2306.04746)
